circuits/main/src/get_account.nr

```diff
@@ -80,27 +80,21 @@ pub(crate) fn assert_account_equals(account_rlp_left_padded: [u8; MAX_ACCOUNT_ST
     );
 }
 
-fn assert_account_proof(account_with_proof: AccountWithProof, state_root: [u8; KEY_LENGTH]) {
+fn assert_account_proof(account_state_proof: StateProof, state_root: [u8; KEY_LENGTH]) {
     let trie_proof = TrieProof {
-        key: account_with_proof.key,
-        proof: account_with_proof.proof,
-        depth: account_with_proof.depth,
-        value: account_with_proof.value
+        key: account_state_proof.key,
+        proof: account_state_proof.proof,
+        depth: account_state_proof.depth,
+        value: account_state_proof.value
     };
     assert(trie_proof.verify_state_root(state_root), "TrieProof: Invalid state root");
 }
 
 pub(crate) fn verify_account(block_no: Field, address: [u8; 20], state_root: [u8; KEY_LENGTH]) -> Account {
-    let account_with_proof = get_account_unconstrained(block_no, address);
+    let (account, state_proof) = get_account_unconstrained(block_no, address);
 
-    let account = Account {
-        nonce: account_with_proof.nonce,
-        balance: account_with_proof.balance,
-        state_root: account_with_proof.state_root,
-        code_hash: account_with_proof.code_hash
-    };
-    assert_account_equals(account_with_proof.value, account);
-    assert_account_proof(account_with_proof, state_root);
+    assert_account_equals(state_proof.value, account);
+    assert_account_proof(state_proof, state_root);
```

Comment:
This is outside of this PR but we also need to verify the connection between `state_root` and `block_hash`